manim/mobject/text/text_mobject.py

```diff
@@ -492,18 +493,58 @@ def __init__(
             if len(each.points) == 0:
                 continue
             points = each.points
-            last = points[0]
-            each.clear_points()
+            curve_start = points[0]
+            assert len(curve_start) == self.dim, curve_start
+            # Some of the glyphs in this text might not be closed,
+            # so we close them by identifying when one curve ends
+            # but it is not where the next curve starts.
+            # It is more efficient to temporarily create a list
+            # of points and add them one at a time, then turn them
+            # into a numpy array at the end, rather than creating
+            # new numpy arrays every time a point or fixing line
+            # is added (which is O(n^2) for numpy arrays).
+            closed_curve_points = []
+            # OpenGL has points be part of quadratic Bezier curves;
+            # Cairo uses cubic Bezier curves.
+            if nppc == 3:
```

Comment:
Just for the sake of maintainability (and if possible), I'd suggest to replace this comparison to `if config.renderer = RendererType.OPENGL`, the other branch to `elif config.renderer == RendererType.CAIRO`, and `raise NotImplementedError(...)` in a new `else`-part.

Reply:
If you think that's a better approach, sure, go ahead and make that change. I wasn't sure if `nppc == 3 iff config.renderer== RendererType.OPENGL`, but I figured if new renderers ever were added, we wouldn't have to change this code assuming `self.n_points_per_curve` was set correctly

Reply:
Good point. Let's keep the comparisons as-is, but add a comment containing `RendererType.CAIRO` / `RendererType.OPENGL` next to it so that once we, e.g., change the cairo renderer to use quadratic curves too, have this function also show up when searching for renderer-dependent code.

Reply:
Done